### ISSUE TITLE
TargetClosedException when using Proxy

### ISSUE THREAD

**Issue by null511** [kblok/puppeteer-sharp#1145]
### Description

A TargetClosedException occurs immediately after the browser window is opened when using proxies with PuppeteerSharp 1.15.0 and 1.16.0; rolling back to 1.14.1 it is working fine. This is occurring both headless and non-headless, in Chrome and Chromium, on Windows and Linux. I am also _not_ using request interception.

### Expected behavior:

Successful proxy usage.

### Actual behavior:

```
Protocol error(Page.navigate): Target closed. (NetworkManager failed to process Fetch.requestPaused. Value cannot be null.
Parameter name: key.    at System.Collections.Concurrent.ConcurrentDictionary`2.ThrowKeyNullException()
   at System.Collections.Concurrent.ConcurrentDictionary`2.set_Item(TKey key, TValue value)
   at PuppeteerSharp.NetworkManager.OnRequestPausedAsync(FetchRequestPausedResponse e)
   at PuppeteerSharp.NetworkManager.Client_MessageReceived(Object sender, MessageEventArgs e)) Protocol error(Page.navigate): Target closed. (NetworkManager failed to process Fetch.requestPaused. Value cannot be null.
Parameter name: key.    at System.Collections.Concurrent.ConcurrentDictionary`2.ThrowKeyNullException()
   at System.Collections.Concurrent.ConcurrentDictionary`2.set_Item(TKey key, TValue value)
   at PuppeteerSharp.NetworkManager.OnRequestPausedAsync(FetchRequestPausedResponse e)
   at PuppeteerSharp.NetworkManager.Client_MessageReceived(Object sender, MessageEventArgs e))
```

### Versions

.NET Core 2.0

**Comment by kblok** [comment]
Hey @null511.
Can you share some code so I can try to replicate the issue?

**Comment by null511** [comment]
My code runs across a few classes, but I've pulled the relevant parts here. I am using https://luminati.io/ API to connect.

```csharp
var launchOptions = new LaunchOptions {
    Headless = true,
    IgnoreHTTPSErrors = true,
    ExecutablePath = filename,
    Args = new [] {
        "--no-sandbox",
        "--disable-infobars",
        "--disable-setuid-sandbox",
        "--ignore-certificate-errors",
        $"--proxy-server=\"<0.0.0.0>:<22225>\"",
    },
};

var browser = await Puppeteer.LaunchAsync(launchOptions);
//browser.TargetCreated += Browser_OnTargetCreated;
//browser.TargetDestroyed += Browser_OnTargetDestroyed;

var DefaultPage = (await browser.PagesAsync())[0];

await DefaultPage.AuthenticateAsync(new Credentials {
    Username = "<user>",
    Password = "<pass>",
});

var opt = new NavigationOptions {
    Referer = "https://www.google.com/",
    WaitUntil = new [] {
        WaitUntilNavigation.Load,
        WaitUntilNavigation.Networkidle0,
    },
};

await DefaultPage.GoToAsync("<url>", opt);
```

**Comment by kblok** [comment]
I created an account on Luminati, navigated to https://www.google.com and it worked.

**Comment by kerohero** [comment]
@kblok I was about to open the same issue, it is a bug 
You can't produce it with Luminati, if you're using their Proxies manager (you won't need auth).
You need to auth so the problem can occur. As @null511 said rolling back to 1.14.1 solves it.

**Comment by kblok** [comment]
Fixed on v1.17.1

**Comment by null511** [comment]
Awesome. I just wanted to confirm that I've updated to 1.17.1 and it appears everything is working fine.

Also, thank you @kerohero that is correct. I did not try the ProxyManager service myself, but did discover that part of the issue above was authentication not being applied somehow.

**Comment by kerohero** [comment]
I can confirm that it's solved too, updated to 1.17.1 and everything runs as expected.
